mapeditor/mainwindow.cpp

```diff
@@ -429,6 +446,87 @@ void MainWindow::on_actionOpen_triggered()
 	openMap(filenameSelect);
 }
 
+void MainWindow::on_actionOpenRecent_triggered()
+{
+	QSettings s(Ui::teamName, Ui::appName);
+	QStringList recentFiles = s.value(recentlyOpenedFilesSetting).toStringList();
+
+	struct RecentFileDialog : public QDialog {
```

Comment:
don't declare such things inside a function, rather place it in anonymous namespace.

also, it should be a class as QDialog is class.

Reply:
struct -> class fixed with 2678bf1, will change from local function to either anonymous namespace or just make it a reusable recent-file dialog with own header in a follow-up commit

Reply:
> will change from local function to either anonymous namespace or just make it a reusable recent-file dialog with own header in a follow-up commit

Before I do this, what's the reason for making it an anonymous class in favor of a local class? The class is not meant to be used anywhere else outside of `on_actionOpenRecent_triggered`.

Reply:
Resolving discussion, please re-open it if you have further remarks.

Reply:
local class makes it more difficult to read a function IMO

Reply:
fair enough, I'll move it out to an anonymous namespace